bitswap/client/client.go

```diff
@@ -176,7 +181,7 @@ func New(parent context.Context, network bsnet.BitSwapNetwork, providerFinder Pr
 
 	if bs.providerFinder != nil && bs.defaultProviderQueryManager {
 		// network can do dialing.
-		pqm, err := rpqm.New(ctx, network, bs.providerFinder, rpqm.WithMaxProviders(10))
+		pqm, err := rpqm.New(ctx, network, bs.providerFinder)
```

Comment:
AFAIK we don't yet have any evidence that this is a better default to set, I wouldn't change the defaults for people unless we have done some tests demonstrating that this is the case.

Note: setting this number to be unbounded might be wasteful since if one request has many providers we might connect to lots of them even though we only need a few, it might be fine in practice though (especially given we have a 5s timeout for the entire query). If we wanted to set the number to be higher / unbounded but were noticing issues related to having too many connections there are probably some more sophisticated approaches we could take instead but better to leave that until it looks necessary 😅

Reply:
The point of this PR is not to change the defaults, but that the default PQM should use its own defaults. So if I understand your comment, then what this PR should also do then is to set `defaultMaxProviders = 10` in `providerquerymanager.go`. Right? 

Or, are you saying that this PQM is the _client's_ default which should have 10 max providers, even though a PQM has a default of unlimited providers?

Reply:
Yeah, we made the PQM-defaults not be the Bitswap-Default-PQM defaults because those were more conservative. I am fine with aligning. Increasing what we had a little bit probably doesn't hurt.